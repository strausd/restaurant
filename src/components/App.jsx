import React from 'react';
import { connect } from 'react-redux';

import AppRouter from '../routers/AppRouter';
import { fetchRestaurants } from '../actions/baseActions'


class App extends React.Component {
    constructor(props) {
        super(props);
        props.fetchRestaurants();
    }

    render() {
        return (
            <div>
                <AppRouter />
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchRestaurants: () => dispatch(fetchRestaurants())
    };
};

export default connect(undefined, mapDispatchToProps)(App);