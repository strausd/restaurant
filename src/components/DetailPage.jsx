import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import RestaurantDetailItem from './RestaurantDetailItem';


export class DetailPage extends React.Component {
    render() {
        const id = this.props.match.params.id;
        if (isNaN(id)) {
            return <Redirect to="/" />
        } else if (!this.props.restaurants || this.props.restaurants.length < 1) {
            return (
                <div className="page-content">
                    Loading...
                </div>
            );
        } else {
            const restaurant = this.props.restaurants[id];
            return (
                <div className="page-content">
                    <RestaurantDetailItem restaurant={restaurant} />
                </div>
            );
        }
    }
}


const mapStateToProps = ({ restaurants, restaurant }) => {
    return {
        restaurants,
        restaurant
    };
};

export default connect(mapStateToProps)(DetailPage);