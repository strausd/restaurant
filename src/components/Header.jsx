import React from 'react';
import { Link } from 'react-router-dom';


const Header = props => {
    const back_button = window.location.pathname !== '/';
    return (
        <div className="header">
            {back_button ? (
                <div className="header__icon header__back">
                    <Link to="/">
                        <img src="/dist/icons/ic_webBack@2x.png" alt="Map" />
                    </Link>
                </div>
            ) : null}
            <div className="header__link">
                <Link to="/">
                    <h1>Lunch Tyme</h1>
                </Link>
            </div>
            <div className="header__icon header__map">
                <Link to="/map">
                    <img src="/dist/icons/icon_map@2x.png" alt="Map" />
                </Link>
            </div>
        </div>
    );
};

export default Header;