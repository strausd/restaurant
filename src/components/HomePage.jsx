import React from 'react';
import { connect } from 'react-redux'

import RestaurantsList from './RestaurantsList';


export const HomePage = () => {
    return (
        <div className="page-content">
            <RestaurantsList />
        </div>
    );
};

export default HomePage;