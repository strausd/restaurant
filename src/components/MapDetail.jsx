import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps"

const MapDetail = withScriptjs(withGoogleMap((props) =>
    <GoogleMap
        defaultZoom={props.zoom}
        defaultCenter={{ lat: props.lat, lng: props.lng }}
    >
        <Marker position={{ lat: props.lat, lng: props.lng }}>
            <InfoWindow>
                <div>Testing</div>
            </InfoWindow>
        </Marker>
    </GoogleMap>
));

export default MapDetail;