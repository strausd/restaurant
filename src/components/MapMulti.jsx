import React from 'react';
import { withScriptjs, withGoogleMap, GoogleMap, Marker, InfoWindow } from "react-google-maps";

export const MapDetail = withScriptjs(withGoogleMap((props) =>
    <GoogleMap
        defaultZoom={props.zoom}
        defaultCenter={{ lat: props.lat, lng: props.lng }}
    >
        {props.restaurants.map((r, index) => {
            return (
                <Marker key={index} position={{ lat: r.lat, lng: r.lng }} onClick={() => props.onMarkerToggle(index)}>
                    {r.isOpenMarkerInfo && <InfoWindow onCloseClick={props.onMarkerToggle}>
                        <div className="marker-info">
                            <div className="marker-info__name">{r.name}</div>
                            <div className="marker-info__directions">
                                <a target="_blank" href={'https://www.google.com/maps/dir/?api=1&destination=' + r.gMapsQuery}>
                                    Get Directions
                                </a>
                            </div>
                        </div>
                    </InfoWindow>}
                </Marker>
            );
        })}
    </GoogleMap>
));

export default MapDetail;