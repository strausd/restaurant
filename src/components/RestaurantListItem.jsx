import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { selectRestaurant } from '../actions/baseActions';


export class RestaurantListItem extends React.Component {
    constructor(props) {
        super(props);
    }

    handleRestaurantClick = () => {
        this.props.selectRestaurant(this.props.restaurant);
    };

    render() {
        return (
            <div className="restaurant-item" id={this.props.index}>
                <Link to={'/' + this.props.index} onClick={this.handleRestaurantClick} >
                    <div className="restaurant-item__info">
                        <h2>{this.props.restaurant.name}</h2>
                        <div className="restaurant-item__category">{this.props.restaurant.category}</div>
                    </div>
                    <div className="restaurant-item__img">
                        <img src={this.props.restaurant.backgroundImageURL} align="middle" />
                    </div>
                </Link>
            </div>
        );
    }
}

const mapDispatchToProps = dispatch => {
    return {
        selectRestaurant: restaurant => dispatch(selectRestaurant(restaurant))
    };
};

export default connect(undefined, mapDispatchToProps)(RestaurantListItem);