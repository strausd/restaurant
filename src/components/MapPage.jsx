import React from 'react';
import { connect } from 'react-redux';

import MapMulti from './MapMulti';
import { getMapInfoArray } from '../misc/utils';


export class MapPage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            restaurant_marker_info: []
        };
    }

    componentWillMount() {
        if (!this.props.restaurants || this.props.restaurants.length < 1) {
            this.setState({ restaurant_marker_info: [] });
        } else {
            const restaurants = getMapInfoArray(this.props.restaurants);
            this.setState({ restaurant_marker_info: restaurants });
        }
    }
    

    componentWillReceiveProps(nextProps) {
        if (!nextProps.restaurants || nextProps.restaurants.length < 1) {
            this.setState({ restaurant_marker_info: [] });
        } else {
            const restaurants = getMapInfoArray(nextProps.restaurants);
            this.setState({ restaurant_marker_info: restaurants });
        }
    }

    onMarkerToggle = (index) => {
        this.setState(prevState => {
            return {
                restaurant_marker_info: prevState.restaurant_marker_info.map((r, r_index) => {
                    if (index === r_index) {
                        return {
                            ...r,
                            isOpenMarkerInfo: !r.isOpenMarkerInfo
                        };
                    } else {
                        return {
                            ...r,
                            isOpenMarkerInfo: false
                        };
                    }
                })
            };
        });
    }

    render() {
        if (!this.props.restaurants || this.props.restaurants.length < 1) {
            return (
                <div className="page-content">
                    Loading...
                </div>
            );
        } else {
            return (
                <div className="page-content">
                    <div className="map-page">
                        <MapMulti 
                            isMarkerShow={true}
                            googleMapURL="https://maps.googleapis.com/maps/api/js?key=AIzaSyBGyVswgcCkctFs8x4X-14BS16VXWINHPA&v=3.exp&libraries=geometry,drawing,places"
                            loadingElement={<div style={{ height: `100%` }} />}
                            containerElement={<div style={{ height: `100%` }} />}
                            mapElement={<div style={{ height: `100%` }} />}
                            lat={32.98}
                            lng={-96.835}
                            zoom={13}
                            onMarkerToggle={this.onMarkerToggle}
                            restaurants={this.state.restaurant_marker_info}
                        />
                    </div>
                </div>
            );
        }
        
    }
}

const mapStateToProps = ({ restaurants }) => {
    return {
        restaurants
    };
};

export default connect(mapStateToProps)(MapPage);