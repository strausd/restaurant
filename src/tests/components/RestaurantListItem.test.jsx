import React from 'react';
import { shallow } from 'enzyme';

import { RestaurantListItem } from '../../components/RestaurantListItem';
import { restaurants } from '../testData/restaurants';


let wrapper, restaurant, selectRestaurant;

beforeAll(() => {
    selectRestaurant = jest.fn();
    restaurant = restaurants[0];
    wrapper = shallow(<RestaurantListItem restaurant={restaurant} index={0} selectRestaurant={selectRestaurant} />);
});

test('should render RestaurantListItem component properly', () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.instance().handleRestaurantClick();
    expect(selectRestaurant).toHaveBeenCalled();
});