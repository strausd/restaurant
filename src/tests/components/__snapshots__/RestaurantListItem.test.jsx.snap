// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RestaurantListItem component properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantListItem
    index={0}
    restaurant={
      Object {
        "backgroundImageURL": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
        "category": "Burgers",
        "contact": Object {
          "formattedPhone": "(972) 387-2337",
          "phone": "9723872337",
          "twitter": "hopdoddy",
        },
        "location": Object {
          "address": "5100 Belt Line Road, STE 502",
          "cc": "US",
          "city": "Addison",
          "country": "United States",
          "crossStreet": "Dallas North Tollway",
          "formattedAddress": Array [
            "5100 Belt Line Road, STE 502 (Dallas North Tollway)",
            "Addison, TX 75254",
            "United States",
          ],
          "lat": 32.950787,
          "lng": -96.821118,
          "postalCode": "75254",
          "state": "TX",
        },
        "name": "Hopdoddy Burger Bar",
      }
    }
    selectRestaurant={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Link
        onClick={[Function]}
        replace={false}
        to="/0"
      >
        <div
          className="restaurant-item__info"
        >
          <h2>
            Hopdoddy Burger Bar
          </h2>
          <div
            className="restaurant-item__category"
          >
            Burgers
          </div>
        </div>
        <div
          className="restaurant-item__img"
        >
          <img
            align="middle"
            src="http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png"
          />
        </div>
      </Link>,
      "className": "restaurant-item",
      "id": 0,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="restaurant-item__info"
          >
            <h2>
              Hopdoddy Burger Bar
            </h2>
            <div
              className="restaurant-item__category"
            >
              Burgers
            </div>
          </div>,
          <div
            className="restaurant-item__img"
          >
            <img
              align="middle"
              src="http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png"
            />
          </div>,
        ],
        "onClick": [Function],
        "replace": false,
        "to": "/0",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Hopdoddy Burger Bar
              </h2>,
              <div
                className="restaurant-item__category"
              >
                Burgers
              </div>,
            ],
            "className": "restaurant-item__info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hopdoddy Burger Bar",
              },
              "ref": null,
              "rendered": "Hopdoddy Burger Bar",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Burgers",
                "className": "restaurant-item__category",
              },
              "ref": null,
              "rendered": "Burgers",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              align="middle"
              src="http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png"
            />,
            "className": "restaurant-item__img",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "align": "middle",
              "src": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Link
          onClick={[Function]}
          replace={false}
          to="/0"
        >
          <div
            className="restaurant-item__info"
          >
            <h2>
              Hopdoddy Burger Bar
            </h2>
            <div
              className="restaurant-item__category"
            >
              Burgers
            </div>
          </div>
          <div
            className="restaurant-item__img"
          >
            <img
              align="middle"
              src="http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png"
            />
          </div>
        </Link>,
        "className": "restaurant-item",
        "id": 0,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="restaurant-item__info"
            >
              <h2>
                Hopdoddy Burger Bar
              </h2>
              <div
                className="restaurant-item__category"
              >
                Burgers
              </div>
            </div>,
            <div
              className="restaurant-item__img"
            >
              <img
                align="middle"
                src="http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png"
              />
            </div>,
          ],
          "onClick": [Function],
          "replace": false,
          "to": "/0",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Hopdoddy Burger Bar
                </h2>,
                <div
                  className="restaurant-item__category"
                >
                  Burgers
                </div>,
              ],
              "className": "restaurant-item__info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Hopdoddy Burger Bar",
                },
                "ref": null,
                "rendered": "Hopdoddy Burger Bar",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Burgers",
                  "className": "restaurant-item__category",
                },
                "ref": null,
                "rendered": "Burgers",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                align="middle"
                src="http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png"
              />,
              "className": "restaurant-item__img",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "align": "middle",
                "src": "http://sandbox.bottlerocketapps.com/BR_iOS_CodingExam_2015_Server/Images/hopdoddy.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
